src/sentry/web/frontend/accounts.py

```diff
@@ -123,7 +123,13 @@ def recover_confirm(request, user_id, hash):
 @login_required
 def start_confirm_email(request):
     has_unverified_emails = request.user.has_unverified_emails()
-    if has_unverified_emails:
+
+    if 'email' in request.GET:
```

Comment:
[Here's](https://github.com/getsentry/sentry/pull/4358/commits/31f5c2673f9f12760a7d81a89efda0f5e331a55f) the commit. This was the behavior we had originally to trigger an email, I just created a separate view for primary email verification. If this is still prone to exploitation, I can change it to use POST.